lib/aruba/environment/windows_environment.rb

```diff
@@ -0,0 +1,11 @@
+require 'aruba/environment/basic_environment'
+
+module Aruba
+  class WindowsEnvironment < BasicEnvironment
+    def initialize
```

Comment:
I did think about that (just changing every ENV name (=key) to uppercase.  But I didn't know if  changes to ENV  would be 'rolled back' after the program ends. So I only wanted to make minimal changes to ENV.  Even though Windows doesn't recognize case, perhaps someone is counting on one of their ENVs being mixed (or whatever) case and is accessing the info some other way. (Through the registry, perhaps.)   
As I started working on a '<code>which</code>' for Windows, I created a method for WindowsEnvironments: given an ENV name, convert it to upper case and save it. Thus we can change (<code>.upcase</code>) only the ENV vars that we really need to deal with.
Whatever way you think is best, I'll make it so. :-)   

> If windows is case-insensitive for ENV, can we savely assume, that the code works for a program written in C, C# or whatever?  

I don't quite understand your question. Which *code* are you referring to?  
Are you asking 'is this the approach that all programs must take when they're dealing with Windows ENVs?'  the answer is yes. (But I don't think that's what you're asking.)